src/JuliusSweetland.OptiKey.Core/GazeSmoothingFilter.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Copyright (c) 2020 OPTIKEY LTD (UK company number 11854839) - All Rights Reserved
+using JuliusSweetland.OptiKey.Models;
+using JuliusSweetland.OptiKey.Properties;
+using System;
+using System.Windows;
+
+namespace JuliusSweetland.OptiKey.DataFilters
+{
+    public class GazeSmoothingFilter
+
+    {
+        private double EstimationNoise;
+        readonly double ProcessNoise;
+        private double Gain;
+        private double SmoothingLevel;
+        private Point EstimatedValue;
+
```

Comment:
Commenting needed:
>>EstimatedPoint: The point that will be returned for use after applying the filter to the delta of the measured point (n) and the previous estimate (n - 1)
Gain: Calculated each iteration and used to scale (0 to 100%) the delta between point (n - 1) and (n) 
SmoothingLevel:  The setting that a user can adjust to influence the filter behavior where a larger value equates to a less responsive more "floaty" cursor
EstimationNoise: Requires a large initial value for (n = 0) but will then fluctuate every iteration 
ProcessNoise: Constant used to fine-tune the filter where a larger value equates to slower movement. Anything from 400 to 3600 seems to work well in this model

Reply:
A bit more precise on a couple of these:

ProcessNoise: the amount of uncertainty in our model of eye movements. This is low when we believe the user is fixating and large when we believe they've moved to a new location.

EstimationNoise: the amount of uncertainty in our current running estimate. This feeds into calculating how much we should weight new data coming in.